src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStep.java

```diff
@@ -122,6 +130,50 @@ public DescriptorImpl() {
             return step;
         }
 
+        /**
+         * Compatibility hack for JENKINS-62305. Only affects runtime behavior of the step, not the snippet generator.
+         * Ideally, password parameters would not be used at all with this step, but there was no documentation or
+         * runtime warnings for this usage previously and so it is relatively common.
+         */
+        @Override
+        public Map<String, Object> customInstantiate(Map<String, Object> map) {
+            if (DescribableModel.of(PasswordParameterValue.class).getParameter("value").getErasedType() != Secret.class) {
+                return map;
+            }
+            return copyMapReplacingEntry(map, "parameters", List.class, parameters -> {
+                List<Object> newParameters = new ArrayList<>(parameters.size());
```

Comment:
`Stream.map` would arguably be more legible here.

Reply:
Yeah, I tried a few approaches and did not really like any of them, happy to switch to streams though.